src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/ProjectRuleSnapshotExtensions.cs

```diff
@@ -4,6 +4,8 @@
 
 #nullable disable
 
+// TODO annotate this file once NotNullIfNotNull is supported https://github.com/dotnet/roslyn/issues/31549
```

Comment:
This file still disables the nullable feature. Without `NotNullIfNotNull` support, it's hard to get this right. https://github.com/dotnet/roslyn/issues/35816 is tracking that work.